#!/usr/bin/env python
# Arips Planner configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("door_handle_approach_x_offset", double_t, 0, "Offset from detected door handle position in door x-direction to calculate approach pose", 0.06, -1.0, 1.0)
gen.add("door_handle_approach_y_offset", double_t, 0, "Offset from detected door handle position in door y-direction to calculate approach pose", 0.01, -1.0, 1.0)
gen.add("servo_angle_for_hooking", double_t, 0, "Kinect servo angle to see the door handle for hooking", -110, -120, -90)
gen.add("drive_open_wall_dist", double_t, 0, "When opening the door while hooked, distance to front wall to stop robot", 0.25, 0.0, 1.0)

exit(gen.generate("arips_navigation", "arips_navigation", "OpenDoor"))
